CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -246,4 +246,33 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
     return Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId];
   }
 
+  /**
+   * check if financial account is referenced by financial item
+   *
+   * @param $financialAccountId Integer
+   *
+  * @param $financialAccountTypeID Integer
+  *
+   * @return bool
+  *
+  */
+ public static function validateFinancialAccount($financialAccountId, $financialAccountTypeID = NULL) {
+   if (!$financialAccountId) {
```

Comment:
It looks like $financialAccountTypeID is not actually an optional parameter - so why not make it required & skip this check?